### PR TITLE
[PM-11303] Add button missing for folders

### PR DESCRIPTION
**Pull request by andrebispo5**
## 🎟️ Tracking

https://bitwarden.atlassian.net/browse/PM-11303

## 📔 Objective

- Added the missing "Add Cipher" floating button to the folder view.
- Updated the "Add" functionality to preselect the current folder when adding from a folder view. To achieve this, a new navigation parameter was introduced and passed to the Add/Edit screen.

## 📸 Screenshots

https://github.com/user-attachments/assets/25db2b95-38ec-4d0d-9440-f13866956830

## ⏰ Reminders before review

- Contributor guidelines followed
- All formatters and local linters executed and passed
- Written new unit and / or integration tests where applicable
- Used internationalization (i18n) for all UI strings
- CI builds passed
- Communicated to DevOps any deployment requirements
- Updated any necessary documentation or informed the documentation team

## 🦮 Reviewer guidelines

- 👍 (`:+1:`) or similar for great changes
- 📝 (`:memo:`) or ℹ️ (`:information_source:`) for notes or general info
- ❓ (`:question:`) for questions
- 🤔 (`:thinking:`) or 💭 (`:thought_balloon:`) for more open inquiry that's not quite a confirmed
  issue and could potentially benefit from discussion
- 🎨 (`:art:`) for suggestions / improvements
- ❌ (`:x:`) or ⚠️ (`:warning:`) for more significant problems or concerns needing attention
- 🌱 (`:seedling:`) or ♻️ (`:recycle:`) for future improvements or indications of technical debt
- ⛏ (`:pick:`) for minor or nitpick changes

### CODE REVIEW

File: app/src/main/java/com/x8bit/bitwarden/ui/vault/feature/addedit/util/CipherViewExtensions.kt [hunk 1]
```diff
@@ -127,7 +127,7 @@ fun VaultAddEditState.ViewState.appendFolderAndOwnerData(
             common = currentContentState.common.copy(
                 selectedFolderId = folderViewList.toSelectedFolderId(
                     cipherView = currentContentState.common.originalCipher,
-                ),
+                ) ?: currentContentState.common.selectedFolderId,
```
Comment (david-livefront) [inline app/src/main/java/com/x8bit/bitwarden/ui/vault/feature/addedit/util/CipherViewExtensions.kt:130]: Can you move this down a line to match the chaining pattern:

```kotlin
)
?: currentContentState.common.selectedFolderId,
```

File: app/src/main/java/com/x8bit/bitwarden/ui/vault/feature/vault/VaultGraphNavigation.kt [hunk 2]
```diff
@@ -20,7 +20,10 @@ const val VAULT_GRAPH_ROUTE: String = "vault_graph"
 @Suppress("LongParameterList")
 fun NavGraphBuilder.vaultGraph(
     navController: NavController,
-    onNavigateToVaultAddItemScreen: (vaultItemCipherType: VaultItemCipherType) -> Unit,
+    onNavigateToVaultAddItemScreen: (
+        vaultItemCipherType: VaultItemCipherType,
+        selectedFolderId: String?,
+        ) -> Unit,
```
Comment (david-livefront) [inline app/src/main/java/com/x8bit/bitwarden/ui/vault/feature/vault/VaultGraphNavigation.kt:26]: Formatter

File: app/src/test/java/com/x8bit/bitwarden/ui/vault/feature/itemlisting/VaultItemListingViewModelTest.kt [hunk 3]
```diff
@@ -942,6 +942,26 @@ class VaultItemListingViewModelTest : BaseViewModelTest() {
             }
         }
 
+    @Suppress("MaxLineLength")
+    @Test
+    fun `AddVaultItemClick inside a folder should emit NavigateToAddVaultItem with a selected folder id`() = runTest {
```
Comment (david-livefront) [inline app/src/test/java/com/x8bit/bitwarden/ui/vault/feature/itemlisting/VaultItemListingViewModelTest.kt:947]: Formatter, there are instructions on how to setup the macro for this [here](https://github.com/bitwarden/android?tab=readme-ov-file#setup)

Comment (andrebispo5) [reply app/src/test/java/com/x8bit/bitwarden/ui/vault/feature/itemlisting/VaultItemListingViewModelTest.kt:947]: Sorry I had it configured already I just forgot about it thinking that the `gradle detekt` would apply our code styling as well.

File: app/src/test/java/com/x8bit/bitwarden/ui/vault/feature/itemlisting/VaultItemListingScreenTest.kt [hunk 4]
```diff
@@ -117,7 +117,9 @@ class VaultItemListingScreenTest : BaseComposeTest() {
                 biometricsManager = biometricsManager,
                 onNavigateBack = { onNavigateBackCalled = true },
                 onNavigateToVaultItem = { onNavigateToVaultItemId = it },
-                onNavigateToVaultAddItemScreen = { onNavigateToVaultAddItemScreenCalled = true },
+                onNavigateToVaultAddItemScreen = { _, _ ->
+                    onNavigateToVaultAddItemScreenCalled = true
```
Comment (david-livefront) [inline app/src/test/java/com/x8bit/bitwarden/ui/vault/feature/itemlisting/VaultItemListingScreenTest.kt:121]: Is there a reason to change this?

Comment (andrebispo5) [reply app/src/test/java/com/x8bit/bitwarden/ui/vault/feature/itemlisting/VaultItemListingScreenTest.kt:121]: otherwise it won't even compile.

Comment (david-livefront) [reply app/src/test/java/com/x8bit/bitwarden/ui/vault/feature/itemlisting/VaultItemListingScreenTest.kt:121]: Ahh, I see the underscores now. 😄

**Review by david-livefront — APPROVED** [review]

**Pull request merged**